File: src/main/java/com/trekmanagement/auth/AuthService.java
package com.trekmanagement.auth;

import com.trekmanagement.auth.dto.*;

public interface AuthService {

    /**
     * Flow 1: Register new user, send verification email.
     * Returns 201 — body contains confirmation message only (no tokens yet).
     */
    void register(RegisterRequest request);

    /**
     * Flow 2: Verify email address using the token from the verification email.
     */
    void verifyEmail(String token);

    /**
     * Flow 3: Authenticate user, issue access + refresh tokens.
     * Requires email_verified = true.
     */
    AuthResponse login(LoginRequest request);

    /**
     * Flow 5: Rotate refresh token — invalidate old, issue new access + refresh pair.
     */
    AuthResponse refresh(String rawRefreshToken);

    /**
     * Flow 6: Revoke refresh token, clear session.
     */
    void logout(String rawRefreshToken);

    /**
     * Flow 7a: Initiate password reset — send reset email.
     * Always returns 200 regardless of whether email exists (ambiguous by design).
     */
    void forgotPassword(String email);

    /**
     * Flow 7b: Complete password reset using token from email.
     */
    void resetPassword(ResetPasswordRequest request);
}

File: extract_tmp2/trek-management-frontend/src/services/authService.ts
import apiClient from '@/api/apiClient'
import type { ApiResponse } from '@/types/api'
import type {
  AuthResponse,
  ForgotPasswordRequest,
  LoginRequest,
  RefreshRequest,
  RegisterRequest,
  ResendVerificationRequest,
  ResetPasswordRequest,
} from '@/types/auth'

const authService = {
  register: (data: RegisterRequest) =>
    apiClient.post<ApiResponse<void>>('/auth/register', data).then((r) => r.data),

  verifyEmail: (token: string) =>
    apiClient.get<ApiResponse<void>>('/auth/verify', { params: { token } }).then((r) => r.data),

  resendVerification: (data: ResendVerificationRequest) =>
    apiClient.post<ApiResponse<void>>('/auth/resend-verification', data).then((r) => r.data),

  login: (data: LoginRequest) =>
    apiClient.post<ApiResponse<AuthResponse>>('/auth/login', data).then((r) => r.data),

  refresh: (data: RefreshRequest) =>
    apiClient.post<ApiResponse<AuthResponse>>('/auth/refresh', data).then((r) => r.data),

  logout: (data: RefreshRequest) =>
    apiClient.post<ApiResponse<void>>('/auth/logout', data).then((r) => r.data),

  forgotPassword: (data: ForgotPasswordRequest) =>
    apiClient.post<ApiResponse<void>>('/auth/forgot-password', data).then((r) => r.data),

  resetPassword: (data: ResetPasswordRequest) =>
    apiClient.post<ApiResponse<void>>('/auth/reset-password', data).then((r) => r.data),
}

export default authService
